components/script/dom/characterdata.rs

```diff
@@ -85,6 +87,16 @@ impl CharacterDataMethods for CharacterData {
         self.content_changed();
         let node = self.upcast::<Node>();
         node.ranges().replace_code_units(node, 0, old_length, new_length);
+
+        // If this is a Text node, we might need to re-parse (say, if our parent
+        // is a <style> element.) We don't need to if this is a Comment or
+        // ProcessingInstruction.
+        if let Some(_) = self.downcast::<Text>() {
```

Comment:
Err, I didn't catch it the first time, but I think this could be `self.is::<Text>()` instead.